// Licensed to the Apache Software Foundation (ASF) under one
// or more contributor license agreements.  See the NOTICE file
// distributed with this work for additional information
// regarding copyright ownership.  The ASF licenses this file
// to you under the Apache License, Version 2.0 (the
// "License"); you may not use this file except in compliance
// with the License.  You may obtain a copy of the License at
//
//   http://www.apache.org/licenses/LICENSE-2.0
//
// Unless required by applicable law or agreed to in writing,
// software distributed under the License is distributed on an
// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
// KIND, either express or implied.  See the License for the
// specific language governing permissions and limitations
// under the License.

syntax = "proto2";

package quickstep.serialization;

import "types/Type.proto";

message TypedValue {
  required Type.TypeID type_id = 1;

  // NOTE(zuyu): For a NULL value, none of the optional fields are filled in.
  optional int32 int_value = 2;
  optional int64 long_value = 3;
  optional float float_value = 4;
  optional double double_value = 5;
  optional bytes out_of_line_data = 6;
  optional int64 datetime_value = 7;
  optional int64 datetime_interval_value = 8;
  optional int64 year_month_interval_value = 9;

  optional uint32 date_value = 10;
}
